internal/services/appservice/helpers/common_web_app_schema.go

```
@@ -0,0 +1,1358 @@
+			storageAccount.AccessKey = *v.AccessKey
+		}
+
+		if v.MountPath != nil {
+			storageAccount.MountPath = *v.MountPath
+		}
+
+		storageAccounts = append(storageAccounts, storageAccount)
+	}
+
+	return storageAccounts
+}
+
+func FlattenConnectionStrings(appConnectionStrings web.ConnectionStringDictionary) []ConnectionString {
+	if len(appConnectionStrings.Properties) == 0 {
+		return nil
+	}
+	var connectionStrings []ConnectionString
+	for k, v := range appConnectionStrings.Properties {
+		connectionString := ConnectionString{
+			Name: k,
+			Type: string(v.Type),
+		}
+		if v.Value != nil {
+			connectionString.Value = *v.Value
+		}
+		connectionStrings = append(connectionStrings, connectionString)
+	}
+
+	return connectionStrings
+}
+
+func ExpandAppSettingsForUpdate(settings map[string]string) *web.StringDictionary {
+	appSettings := make(map[string]*string)
+	for k, v := range settings {
+		appSettings[k] = pointer.To(v)
+	}
+
+	return &web.StringDictionary{
+		Properties: appSettings,
+	}
+}
+
+func ExpandAppSettingsForCreate(settings map[string]string) *[]web.NameValuePair {
+	if len(settings) > 0 {
+		result := make([]web.NameValuePair, 0)
+		for k, v := range settings {
+			result = append(result, web.NameValuePair{
+				Name:  pointer.To(k),
+				Value: pointer.To(v),
+			})
+		}
+		return &result
+	}
+	return nil
+}
+
+func FlattenAppSettings(input web.StringDictionary) (map[string]string, *int) {
+	maxPingFailures := "WEBSITE_HEALTHCHECK_MAXPINGFAILURES"
+	unmanagedSettings := []string{
+		"DIAGNOSTICS_AZUREBLOBCONTAINERSASURL",
+		"DIAGNOSTICS_AZUREBLOBRETENTIONINDAYS",
+		"WEBSITE_HTTPLOGGING_CONTAINER_URL",
+		"WEBSITE_HTTPLOGGING_RETENTION_DAYS",
+		"WEBSITE_VNET_ROUTE_ALL",
+		"spring.datasource.password",
+		"spring.datasource.url",
+		"spring.datasource.username",
+		maxPingFailures,
+	}
+
+	var healthCheckCount *int
+	appSettings := FlattenWebStringDictionary(input)
+	if v, ok := appSettings[maxPingFailures]; ok {
+		h, _ := strconv.Atoi(v)
```

we should raise this error